### PR TITLE
Cve lookup plugin

### PR DESCRIPTION
**Pull request by mellowCS**

### CODE REVIEW

**Comment by dorpvom** [comment]
Generally, this will not be merged as long as better code hub will downgrade our Code Quality. So have a look at the issues that BCH has with your code

File: src/plugins/analysis/cve_lookup/code/vuln_lookup_plugin.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    :return: tuple of list containing product search terms and the unbound version string
+    '''
+    product_terms = product_component.split()
+    product_search_terms = ['_'.join(product_terms[i:j]).lower() for i, j in
+                            combinations(range(len(product_terms) + 1), 2)]
+    product_search_terms = unbinding(product_search_terms)
+
+    unbound_version = unbinding([input_version])
+
+    return product_search_terms, unbound_version
```
[the remark this answers is not in the document]
Comment (mellowCS) [reply src/plugins/analysis/cve_lookup/code/vuln_lookup_plugin.py]: Done

File: src/plugins/analysis/cve_lookup/internal/setup_repository.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    if args.start_year < 2002 or args.start_year > current_year:
+        raise ValueError('ERROR: Value of \'start_year\' out of bounds. '
+                         'Look at setup_repository.py -h for more information.')
+    if args.end_year < 2002 or args.end_year > current_year:
+        raise ValueError('ERROR: Value of \'end_year\' out of bounds. '
+                         'Look at setup_repository.py -h for more information.')
+    if args.start_year > args.end_year:
+        raise ValueError('ERROR: Value of \'start_year\' greater than value of \'end_year\'.')
+
+    init_rep(args.db_name, args.update, args.specify, args.start_year, args.end_year, args.extraction_path)
```
[the remark this answers is not in the document]
Comment (mellowCS) [reply src/plugins/analysis/cve_lookup/internal/setup_repository.py]: Done

File: src/plugins/analysis/cve_lookup/internal/data_prep.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    if all_cve_urls is None:
+        exit('Error: No CVE links are provided')
+    if years is None:
+        exit('Error: The required years of CVE feeds are not specified')
+    cve_candidates = [url for url in all_cve_urls for year in years if str(year) in url]
+
+    return cve_candidates
+
+
+def download_data(cpe: bool = False, cve: bool = False, update: bool = False, path: str = None,
```
[the remark this answers is not in the document]
Comment (mellowCS) [reply src/plugins/analysis/cve_lookup/internal/data_prep.py]: Done

File: src/plugins/analysis/cve_lookup/internal/meta.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+
+def get_meta(specified_file: str = None) -> dict:
+    '''
+    Retrieves data from json file
+    :param specified_file: path to specified json file
+    :return: json content in dictionary
+    '''
+    if not specified_file:
+        meta_path = path[0] + '/plugins/analysis/cve_lookup/internal/metadata.json'
```
[the remark this answers is not in the document]
Comment (mellowCS) [reply src/plugins/analysis/cve_lookup/internal/meta.py]: Done

File: src/plugins/analysis/cve_lookup/code/vuln_lookup_plugin.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+
+def init_lookup(product_name: str, requested_version: str) -> list:
+    '''
+    gets search terms and initiates functions to find the vulnerabilities for the search terms
+    :param product_name: product name from software components' summary
+    :param requested_version: version from software components' summary
+    :return list containing CVE candidates found with vendor -, product name and version from CPE dictionary
+    '''
+    meta = get_meta()
```
[the remark this answers is not in the document]
Comment (mellowCS) [reply src/plugins/analysis/cve_lookup/code/vuln_lookup_plugin.py]: Done

File: src/plugins/analysis/cve_lookup/code/vuln_lookup_plugin.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    with DB('cpe_cve.db') as db:
+        product_terms, version = generate_search_terms(product_name, requested_version)
+        vendor, product, version = cpe_matching(db, meta, product_terms, version)
+        cve_candidates = cve_cpe_search(db, meta, vendor, product, version)
```
[the remark this answers is not in the document]
Comment (mellowCS) [reply src/plugins/analysis/cve_lookup/code/vuln_lookup_plugin.py]: Done

**Pull request closed without merging**